### PR TITLE
[Merged by Bors] - feat: port Archive.Imo.Imo2005Q4

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Archive/Imo/Imo2005Q4.lean [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  intro h
+  -- Conversely, suppose `k` is a number with the property, and let `p` be `k.min_fac` (by
+  -- definition this is the minimal prime factor of `k` if `k ≠ 1`, and otherwise `1`.
+  let p := k.minFac
+  -- Suppose for the sake of contradiction that `k ≠ 1`.  Then `p` is genuinely a prime factor of
+  -- `k`. Hence, it divides none of `a n`, `1 ≤ n`
+  by_contra hk'
+  have hp : Nat.Prime p := Nat.minFac_prime hk'
+  replace h : ∀ n, 1 ≤ n → ¬(p : ℤ) ∣ a n := fun n hn ↦ by
+    have : IsCoprime (a n) p := .of_isCoprime_of_dvd_right (h n hn) (Int.coe_nat_dvd.mpr k.minFac_dvd)
```
Comment (urkud) [inline Archive/Imo/Imo2005Q4.lean:75]:
```suggestion
    have : IsCoprime (a n) p :=
      .of_isCoprime_of_dvd_right (h n hn) (Int.coe_nat_dvd.mpr k.minFac_dvd)
```

**Comment by hrmacbeth** [comment]
@urkud If you don't mind, I'd rather not golf this problem in the way you did in 243193a (moving from `Int.ModEq` on `Int` to equality on `ZMod`).  Rather, I'd like to keep it in `Int.ModEq` and golf using `gcongr` -- this problem was one of the test cases I had in mind for `gcongr` when I wrote it.

**Comment by urkud** [comment]
I like the `1/2 + 1/3 + 1/6 - 1 = 0` computation. Is it OK if I leave both versions?

**Comment by hrmacbeth** [comment]
Sure!

**Comment by semorrison** [comment]
This appears to be the last IMO file to port. @urkud, would it be okay to mark this as `help-wanted` if you're not planning on looking at this soon?

**Comment by urkud** [comment]
If i don't do it today, then I'll mark it as help wanted.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**